pandas/core/reshape/merge.py

```diff
@@ -1311,7 +1313,16 @@ def _create_cross_configuration(
             DataFrames with cross_col, the merge operation set to inner and the column
             to join over.
         """
-        cross_col = f"_cross_{hashlib.md5().hexdigest()}"
+        _md5: Callable
+        if PY39:
+
+            def _md5(_hashlib_md5=hashlib.md5):
+                return _hashlib_md5(usedforsecurity=False)
+
+        else:
+            _md5 = hashlib.md5
+
+        cross_col = f"_cross_{_md5().hexdigest()}"
```

Comment:
Curious, why was md5 chosen originally? (Looks like you were the original author https://github.com/pandas-dev/pandas/pull/37864 ;))

If the intention was to generate a unique column for joining later, could we just use `uuid.uuid4()` instead?

Reply:
Yes, the intention was generating a unique column, not sure anymore why md5 explicitly. Can change I think